dali/operators/reader/loader/video/video_file.cc

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/video/video_file.h"
+#include "dali/core/error_handling.h"
+
+namespace dali {
+
+void VideoFileCPU::InitAvState() {
+  codec_ctx_ = avcodec_alloc_context3(codec_);
+  DALI_ENFORCE(codec_ctx_, "Could not create av codec context");
+
+  DALI_ENFORCE(
```

Comment:
I wonder how relevant these error messages are.

Reply:
I added error messages parsed from return codes, where I could, so it should improve the situation.
I would rather avoid printing `codec_` or `codec_params_` since what is available in them at any given moment is rather fluid.